### ISSUE TITLE
Memory footprint of Memgraph Platform is not reduced

### ISSUE THREAD

**Issue by kgolubic** [memgraph/memgraph#1552]
**Memgraph version**
Memgrpah Platform - 2.12.1-memgraph2.12.1-lab2.9.1-mage1.12.1

**Environment**
Docker, Windows 11.

**Describe the bug**
The memory footprint of Memgraph Platform is not reduced after the start, but if memory is filled all the way up then it is released.

**To Reproduce**
1. Run the Memgraph Platform image
2. Connect via Lab 
3. Observe the memory over time

**Expected behavior**
Memory should be released after some time.

**Logs**

The clean run has the same memory footprint on the start and after being idle for some time.

![image](https://github.com/memgraph/memgraph/assets/46486712/c141894c-46d9-4456-88a2-c43f0b77ca45)

If a query is run that fills up the memory after the query is terminated the footprint is smaller than with a clean instance.

![image](https://github.com/memgraph/memgraph/assets/46486712/ac00e062-61b3-4cfb-9896-f9d730cf7164)

![image](https://github.com/memgraph/memgraph/assets/46486712/c4ca59ff-8113-4deb-8fe1-44fefb35bf7c)

I've loaded the dataset using:

```
LOAD CSV FROM "https://public-assets.memgraph.com/berlin-subway/berlin-stations.csv" WITH HEADER AS row
CREATE (n:Station {name: row.station, lat: toFloat(row.lat), lng: toFloat(row.lng), end: toInteger(row.end )});
LOAD CSV FROM "https://public-assets.memgraph.com/berlin-subway/berlin-lines.csv" WITH HEADER AS row
MATCH (s1:Station {name: row.station1}), (s2:Station {name: row.station2})
CREATE (s1)-[:CONNECTED_VIA {line: row.line, time: ToInteger(row.time)}]->(s2);
```

And then I run the query:

```
MATCH path=(s1:Station {name: "Uhlandstraße"})-[*]-(s2:Station {name: "Hallesches Tor"}) 
RETURN path;
```

**Additional context**

**Verification Environment**
Once we fix it, what do you need to verify the fix?
Do you need:

* Full Memgraph platform

**Comment by DavIvek** [comment]
It looks like this might be fixed (it couldn't reproduce it using the 2.18 version of Memgraph). A lot of work was put into fixing memory issues in the last couple of releases, so this might be solved. 
@kgolubic are you still experiencing this kind of issue?

**Comment by kgolubic** [comment]
@DavIvek I tried to reproduce it, and I couldn't. I've used 2.18. I don't have this issue any longer.
